File: frontend/lib/core/constants/notification_source.dart
/// Notification source enum
enum NotificationSource {
  claude,
  codex,
  slack,
  github,
  gmail,
}

extension NotificationSourceExtension on NotificationSource {
  String get displayName {
    switch (this) {
      case NotificationSource.claude:
        return 'Claude';
      case NotificationSource.codex:
        return 'Codex';
      case NotificationSource.slack:
        return 'Slack';
      case NotificationSource.github:
        return 'GitHub';
      case NotificationSource.gmail:
        return 'Gmail';
    }
  }

  String get apiValue {
    switch (this) {
      case NotificationSource.claude:
        return 'CLAUDE';
      case NotificationSource.codex:
        return 'CODEX';
      case NotificationSource.slack:
        return 'SLACK';
      case NotificationSource.github:
        return 'GITHUB';
      case NotificationSource.gmail:
        return 'GMAIL';
    }
  }

  static NotificationSource fromApiValue(String value) {
    switch (value.toUpperCase()) {
      case 'CLAUDE':
        return NotificationSource.claude;
      case 'CODEX':
        return NotificationSource.codex;
      case 'SLACK':
        return NotificationSource.slack;
      case 'GITHUB':
        return NotificationSource.github;
      case 'GMAIL':
        return NotificationSource.gmail;
      default:
        throw ArgumentError('Invalid notification source: $value');
    }
  }
}

File: backend/src/main/java/com/notio/notification/domain/NotificationSource.java
package com.notio.notification.domain;

import com.notio.common.exception.ErrorCode;
import com.notio.common.exception.NotioException;

public enum NotificationSource {
    CLAUDE,
    CODEX,
    SLACK,
    GITHUB,
    GMAIL,
    INTERNAL;

    public static NotificationSource from(final String value) {
        try {
            return NotificationSource.valueOf(value.trim().toUpperCase());
        } catch (RuntimeException exception) {
            throw new NotioException(
                    ErrorCode.INVALID_REQUEST,
                    "지원하지 않는 source 입니다."
            );
        }
    }
}
